### PR TITLE
[aotinductor] support at::convolution for AOTInductor

### PR DESCRIPTION
**Pull request by chenyang78**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114961

This PR adds support to at::convolution for AOTInductor

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/csrc/inductor/aoti_torch/shim_common.cpp [hunk 1]
```diff
@@ -292,6 +302,47 @@ AOTITorchError aoti_torch__scaled_dot_product_flash_attention(
   });
 }
 
+AOTI_TORCH_EXPORT AOTITorchError aoti_torch_convolution(
```
Comment (eellison) [inline torch/csrc/inductor/aoti_torch/shim_common.cpp:305]: Would it make sense to codegen these ?

Comment (desertfire) [reply torch/csrc/inductor/aoti_torch/shim_common.cpp:305]: I think @eellison meant auto-generating shim functions. There were not many interfaces needed initially. Hopefully with more complete decomposition coverage, MM and Conv are the only cases we need to have in the C shim layer.

File: torch/_inductor/codegen/wrapper.py [hunk 2]
```diff
@@ -1729,11 +1732,28 @@ def generate_c_shim_extern_kernel_call(self, kernel, args):
         shim_fn = f"aoti_torch_{kernel_suffix}"
         self.writeline(f"AOTI_TORCH_ERROR_CODE_CHECK({shim_fn}({', '.join(args)}));")
 
-    def generate_c_shim_extern_kernel_alloc_call(self, extern_kernel, args):
+    def generate_c_shim_extern_kernel_alloc(self, extern_kernel, args):
```
Comment (desertfire) [inline torch/_inductor/codegen/wrapper.py:1735]: We could use some refactoring for `ir.ExternKernelAlloc` and `ir.FallbackKernel` at some point.

File: torch/csrc/inductor/aoti_torch/c/shim.h [hunk 3]
```diff
@@ -194,6 +194,23 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch__scaled_dot_product_flash_attention(
...
+    AtenTensorHandle input,
+    AtenTensorHandle weight,
+    AtenTensorHandle bias, // optional argument
+    int64_t* stride_ptr,
+    int64_t stride_size,
+    int64_t* padding_ptr,
+    int64_t padding_size,
+    int64_t* dilation_ptr,
+    int64_t dilation_size,
+    bool transposed,
```
Comment (desertfire) [inline torch/csrc/inductor/aoti_torch/c/shim.h:207]: use `int` instead

**Review by desertfire — APPROVED** [review]

File: torch/csrc/inductor/aoti_torch/c/shim.h [hunk 4]
```diff
@@ -233,6 +233,23 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch__scaled_mm(
...
+    AtenTensorHandle weight,
+    AtenTensorHandle bias, // optional argument
+    int64_t* stride_ptr,
+    int64_t stride_size,
+    int64_t* padding_ptr,
+    int64_t padding_size,
+    int64_t* dilation_ptr,
+    int64_t dilation_size,
+    int transposed,
+    int64_t* output_padding_ptr,
```
Comment (swolchok) [inline torch/csrc/inductor/aoti_torch/c/shim.h:247]: I think all these pointers needed to be decorated with `const`. piggybacking fix onto #113577

**Pull request closed without merging**